### PR TITLE
nRF RPC: Protocol versioning

### PR DESCRIPTION
**Pull request by KAGA164**
This PR introduces protocol versioning for the nRF RPC library.
The protocol version range is exchanged  and checked between remote processor during the group initialization process.

### CODE REVIEW

File: nrf_rpc/nrf_rpc.c [hunk 1]
```diff
@@ -237,6 +264,45 @@ static int simple_send(const struct nrf_rpc_group *group, uint8_t dst, uint8_t t
...
+
+	len = strlen(group->strid) + NRF_RPC_PROTOCOL_VERSION_FIELD_SIZE;
+
+	nrf_rpc_alloc_tx_buf(group, &tx_buf, len);
+	if (tx_buf == NULL) {
+		return -NRF_ENOMEM;
+	}
+
+	packet = tx_buf;
+	tx_buf -= NRF_RPC_HEADER_SIZE;
```
Comment (maje-emb) [inline nrf_rpc/nrf_rpc.c:289]: Please add a comment explaining that buffer allocation includes header space.

File: nrf_rpc/nrf_rpc.c [hunk 2]
```diff
@@ -370,40 +436,111 @@ static void execute_packet(const uint8_t *packet, size_t len)
...
+	const struct nrf_rpc_group *group;
 
 	for (NRF_RPC_AUTO_ARR_FOR(iter, group, &nrf_rpc_groups_array,
 				 const struct nrf_rpc_group)) {
-		if ((remote_strid_len == strlen(group->strid)) &&
-		    (memcmp(remote_strid, group->strid, remote_strid_len) == 0)) {
-			group->data->dst_group_id = hdr->src_group_id;
+		if ((strid_len == strlen(group->strid)) &&
+		    (memcmp(strid, group->strid, strid_len) == 0)) {
+			group->data->dst_group_id = src_group_id;
```
Comment (maje-emb) [inline nrf_rpc/nrf_rpc.c:460]: Do I understand `nrf_rpc_groups_array` contains group names with termination character, and the transmitted `strid` is without termination char?
Is it not possible to transmit a `strid` with termination character? Then it is possible to use the functions for string operations.

**Review by maje-emb — COMMENTED** [review]
Looks good.
A few minor comments.

File: nrf_rpc/nrf_rpc.c [hunk 3]
```diff
@@ -20,6 +20,12 @@
 /* Internal definition used in the macros. */
 #define NRF_RPC_HEADER_SIZE 5
 
+/* Protocol version field size inside initialization packet. */
+#define NRF_RPC_PROTOCOL_VERSION_FIELD_SIZE 1
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply nrf_rpc/nrf_rpc.c]: Good idea :) Done

File: nrf_rpc/nrf_rpc.c (hunk 1 above)
Comment (KAGA164) [reply nrf_rpc/nrf_rpc.c]: Sure. Added

**Review by kapi-no — APPROVED** [review]
ACK for rpc versioning commit

**Comment by KAGA164** [comment]
I rebased PR on current main branch

**Review by grochu — APPROVED** [review]
Looks good.

**Review by doki-nordic — APPROVED** [review]

**Pull request merged**